sfdc-accounts-to-dust.ts

```
@@ -0,0 +1,422 @@
+    console.log(`Upserted account ${account.Id} to Dust datasource`);
+  } catch (error) {
+    console.error(
+      `Error upserting account ${account.Id} to Dust datasource:`,
+      error
+    );
+  }
+}
+
+async function upsertTableRowsToDustDatasource(accounts: Account[]) {
+  const rows = accounts.map((account) => ({
+    row_id: `account-${account.Id}`,
+    value: {
+      company_name: account.Name,
+      industry: account.Industry || "N/A",
+      arr: account.AnnualRevenue ? account.AnnualRevenue : -1,
+      number_of_employees: account.NumberOfEmployees || "N/A",
+      type: account.Type || "N/A",
+      rating: account.Rating || "N/A",
+      source: account.AccountSource || "N/A",
+      created_date: account.CreatedDate,
+      last_modified_date: account.LastModifiedDate,
+      last_activity_date: account.LastActivityDate || "N/A",
+    },
+  }));
+
+  await rateLimitedDustApiPost(
+    `/w/${DUST_WORKSPACE_ID}/vaults/${DUST_VAULT_ID}/data_sources/${DUST_DATASOURCE_ID}/tables/${DUST_TABLE_ID}`,
+    {
+      rows,
+      truncate: false,
+    }
+  );
+}
+
+async function main() {
+  try {
+    const conn = await connectToSalesforce();
+
+    console.log("Fetching recently updated accounts from Salesforce...");
+    const accountIds = await getRecentlyUpdatedAccountIds(conn);
+    console.log(
+      `Found ${accountIds.size} accounts with updates in the last 24 hours.`
+    );
+
+    const accounts = await getAccountDetails(conn, Array.from(accountIds));
+    await conn.logout();
+
+    console.log(`Processing ${accounts.length} accounts...`);
+    const upsertPromises = accounts.map((account) =>
+      upsertDocumentToDustDatasource(account)
+    );
+
+    if (IMPORT_AS_TABLE) {
+      // Create the table if it does not exist.
+      const existingTable = await dustApi.get(
+        `/w/${DUST_WORKSPACE_ID}/vaults/${DUST_VAULT_ID}/data_sources/${DUST_DATASOURCE_ID}/tables/${DUST_TABLE_ID}`
+      );
+
+      if (existingTable.status === 404) {
```

[EH-003] Don't use empty catch blocks. At minimum, log the error or handle it appropriately to maintain traceability.